### DISCUSSION TITLE
External mesh file - merging blocks?

### DISCUSSION THREAD

**Discussion by aiskhak** [idaholab/moose#19071] [Q&A Meshing] [answered]
Hello everyone,

I am importing mesh in a moose-based application (SAM) for 3D fluid dynamics modeling. The mesh is composed of different blocks. It seems like SAM treats the boundaries between the blocks as walls.

The geometry and mesh are generated using .geo file. The surfaces are merged according to BCs:

```
Physical Surface("inlet") = {102,106,109,112,114};
Physical Surface("outlet") = {8,31,54,77};
Physical Surface("wall") = {7,9,12,18,20,28,30,32,35,41,43,51,53,55,58,64,66,74,76,78,80,85,86,92,104,108,111,113,153,157,160,163,165};
```

The volume (whole) is merged as:
`Physical Volume("fluid") = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44};`

The remaining surfaces (between the blocks) are not "touched". Then the .geo file opened by gmsh and saved as .msh file:

```
Mesh.ElementOrder = 1;​ 
Mesh.MshFileVersion = 2.16;​
Mesh 3;​
Save "tamu.msh";
```

When I get the results it seems like the surfaces between the blocks are treated as walls and fluid cannot penetrate them. How to fix that?

**Comment by GiudGiud** [comment]
@travismui

**Reply by GiudGiud** [reply]
@rui-hu
do you know people who have experience with using SAM and .geo (gmsh files)

**Reply by travismui** [reply]
Hi @GiudGiud, sorry for the radio silence. We're in the process of resolving this internally, @aiskhak is actually one of our affiliate students. It seems like this was a SAM-related input syntax issue, if we find any relevant info for the larger community we'll try to share it here.

**Comment by aiskhak** [comment, marked as answer]
Sorry for not reporting. The problem is caused by sharp mesh transition regions. The generation of unstructured grids without sharp transitions resolved the issue. Thanks, everyone.
